cmd/lncli/commands.go

```diff
@@ -2355,6 +2362,30 @@ func sendPaymentRequest(ctx *cli.Context, req *lnrpc.SendRequest) error {
 
 	req.AllowSelfPayment = ctx.Bool("allow_self_payment")
 
+	// Parse custom data records.
+	data := ctx.String(dataFlag.Name)
+	if data != "" {
+		records := strings.Split(data, ",")
+		for _, r := range records {
+			kv := strings.Split(r, "=")
+			if len(kv) != 2 {
+				return errors.New("invalid data format")
+			}
+
+			recordID, err := strconv.ParseUint(kv[0], 10, 64)
+			if err != nil {
+				return errors.New("invalid data format")
```

Comment:
error messages could be more descriptive, and describe why it’s invalid

Reply:
fixed